src/include/platform/DeviceInstanceInfoProvider.h

```diff
@@ -0,0 +1,115 @@
+    virtual ~DeviceInstanceInfoProvider() = default;
+
+    /**
+     * @brief Obtain the Serial Number from the device's factory data.
+     *
+     * The SerialNumber attribute specifies a human readable serial number
+     *
+     * @param[in, out] buf Buffer to copy string. This buffer must be null-terminated.
+     * @param[in] bufSize Size of null-terminated input buffer. This size should be +1 higher than maximum possible string.
+     * @returns CHIP_NO_ERROR on success, or another CHIP_ERROR from the underlying implementation
+     *          if access fails.
+     */
+    virtual CHIP_ERROR GetSerialNumber(char * buf, size_t bufSize) = 0;
+
+    /**
+     * @brief Obtain a manufacturing date from the device's factory data.
+     *
+     * The ManufacturingDate attribute specifies the date that the Node was manufactured.
+     * The date format is ISO 8601 YYYYMMDD.
+     *
+     *
+     * @param[in,out] year Reference to location where manufacturing year will be stored
+     * @param[in, out] month Reference to location where manufacturing month will be stored
+     * @param[in, out] day Reference to location where manufacturing day will be stored
+     * @returns CHIP_NO_ERROR on success, or another CHIP_ERROR from the underlying implementation
+     *          if access fails.
+     */
+    virtual CHIP_ERROR GetManufacturingDate(uint16_t & year, uint8_t & month, uint8_t & day) = 0;
+
+    /**
+     * @brief Obtain a Firmware Version from the device's factory data.
+     *
+     * @param[in,out] hardwareVersion Reference to location where the hardware version integer will be copied
+     * @returns CHIP_NO_ERROR on success, or another CHIP_ERROR from the underlying implementation
+     *          if access fails.
+     */
+    virtual CHIP_ERROR GetHardwareVersion(uint16_t & hardwareVersion) = 0;
+
+    /**
+     * @brief Obtain a Firmware Version String from the device's factory data.
+     *
+     * The HardwareVersionString can be used to provide a more user-friendly value than that
+     * represented by the HardwareVersion attribute.
+     *
+     * @param[in, out] buf Buffer to copy string. This buffer must be null-terminated.
+     * @param[in] bufSize Size of null-terminated input buffer. This size should be +1 higher than maximum possible string.
+     * @returns CHIP_NO_ERROR on success, or another CHIP_ERROR from the underlying implementation
+     *          if access fails.
+     */
+    virtual CHIP_ERROR GetHardwareVersionString(char * buf, size_t bufSize) = 0;
+
+    /**
+     * @brief Obtain a Rotating Device ID Unique ID from the device's factory data.
+     *
+     * The unique identifier consists of a randomly-generated 128-bit or longer octet string which
+     * was programmed during factory provisioning or delivered to the device by the vendor using
+     * secure means after a software update
+     *
+     * @param[in,out] uniqueIdSpan Reference to location where the Rotating Device ID Unique ID will be copied
```

Comment:
Will this function update the size of the bytespan?  Is it expecting a cerain size on input?